packages/shared/src/server/repositories/clickhouse.ts

```diff
@@ -1,51 +1,56 @@
+import Decimal from "decimal.js";
+import { z } from "zod";
+import { JsonNested } from "../../utils/zod";
 import {
-  parseJsonPrioritised,
-  type JsonNested,
-  type ObservationType,
-  type ObservationLevel,
-  type Trace,
-  type ObservationView,
-} from "@langfuse/shared";
+  ObservationLevel,
+  ObservationType,
+  ObservationView,
+  Trace,
+} from "@prisma/client";
+import { parseJsonPrioritised } from "../../utils/json";
+import { env } from "../../env";
 import {
-  clickhouseClient,
   clickhouseStringDateSchema,
   observationRecordReadSchema,
   scoreRecordReadSchema,
   traceRecordReadSchema,
-} from "@langfuse/shared/src/server";
-import Decimal from "decimal.js";
-import { z } from "zod";
+} from "../definitions";
+import { clickhouseClient } from "../clickhouse/client";
+import { logger } from "../logger";
+import { getCurrentSpan } from "../instrumentation";
 
 export const getObservation = async (
   observationId: string,
-  projectId: string,
+  projectId: string
 ) => {
-  const query = `SELECT * FROM observations FINAL WHERE id = '${observationId}' AND project_id = '${projectId}' LIMIT 1`;
-  const records = await queryClickhouse(query);
+  const query = `SELECT * FROM observations FINAL WHERE id = {observationId: String} AND project_id = {projectId: String} LIMIT 1`;
+  const records = await queryClickhouse({
+    query,
+    params: { observationId, projectId },
+  });
 
   return records.length ? convertObservations(records)[0] : undefined;
```

Comment:
Is records empty if there is no result? Then `return convertObservations(records).shift()` is way easier to read.